src/cli/cli_tcat.cpp

```diff
@@ -0,0 +1,174 @@
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "openthread-core-config.h"
+
+#include "cli/cli_output.hpp"
+
+#include "cli/cli_tcat.hpp"
+
+#include <openthread/ble_secure.h>
+
+#include <mbedtls/oid.h>
+#include <openthread/tcat.h>
+#include <openthread/platform/ble.h>
+
+#if OPENTHREAD_CONFIG_BLE_TCAT_ENABLE && OPENTHREAD_CONFIG_BLE_SECURE_CLI_ENABLE
+
+#define OT_CLI_BBTC_X509_CERT                                              \
+    "-----BEGIN CERTIFICATE-----\r\n"                                      \
+    "MIIBdzCCAR2gAwIBAgIEESIzAzAKBggqhkjOPQQDAjBaMQswCQYDVQQGEwJERTER\r\n" \
+    "MA8GA1UEBxMIR2FyY2hpbmcxDDAKBgNVBAsTA1NUQTERMA8GA1UEChMITXlWZW5k\r\n" \
+    "b3IxFzAVBgNVBAMTDm9wdG90cm9uaWMuY29tMB4XDTIzMDUyMDIzMDYxNloXDTI0\r\n" \
+    "MDUyMDIzMDYxNlowFTETMBEGA1UEAxMKRGV2aWNlVHlwZTBZMBMGByqGSM49AgEG\r\n" \
+    "CCqGSM49AwEHA0IABNmItKyl/hfcUXALCehmka0iO7sZtRUcLIqL2zbD6X+2EjY7\r\n" \
+    "sInir9hSIyf7V+doP/Z+W5v8yG3402jJNk47ZqOjFjAUMBIGCSsGAQQBgt8qAwQF\r\n" \
+    "AAA/AAAwCgYIKoZIzj0EAwIDSAAwRQIhAPrqeTxnxkDbA/lDpvozNsc5UPeY6xt5\r\n" \
+    "kRjRj73CUN3tAiBycioHdhXOx8+f7fYe030Nw5wzDgjJ8wuu/QElIeCzCg==\r\n"     \
+    "-----END CERTIFICATE-----\r\n"
+
+#define OT_CLI_BBTC_PRIV_KEY                                               \
+    "-----BEGIN EC PRIVATE KEY-----\r\n"                                   \
+    "MHcCAQEEIF4xKKPlATLttEC2OOvzjKSFUo85tE7uh68vdn0hj965oAoGCCqGSM49\r\n" \
+    "AwEHoUQDQgAE2Yi0rKX+F9xRcAsJ6GaRrSI7uxm1FRwsiovbNsPpf7YSNjuwieKv\r\n" \
+    "2FIjJ/tX52g/9n5bm/zIbfjTaMk2Tjtmow==\r\n"                             \
+    "-----END EC PRIVATE KEY-----\r\n"
+
+#define OT_CLI_BBTC_TRUSTED_ROOT_CERTIFICATE                               \
+    "-----BEGIN CERTIFICATE-----\r\n"                                      \
+    "MIIB3TCCAYOgAwIBAgIJAIEkU9Kpk7sQMAoGCCqGSM49BAMCMFoxCzAJBgNVBAYT\r\n" \
+    "AkRFMREwDwYDVQQHEwhHYXJjaGluZzEMMAoGA1UECxMDU1RBMREwDwYDVQQKEwhN\r\n" \
+    "eVZlbmRvcjEXMBUGA1UEAxMOb3B0b3Ryb25pYy5jb20wHhcNMjMwMzI0MjMwODI2\r\n" \
+    "WhcNMjYwMzI0MjMwODI2WjBaMQswCQYDVQQGEwJERTERMA8GA1UEBxMIR2FyY2hp\r\n" \
+    "bmcxDDAKBgNVBAsTA1NUQTERMA8GA1UEChMITXlWZW5kb3IxFzAVBgNVBAMTDm9w\r\n" \
+    "dG90cm9uaWMuY29tMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEIZXjlDNlAxIV\r\n" \
+    "k19EVfeQRj755MWWlZnDhaZKbMPuuP+EML9zdIwWDeCleRP5tKq5fmWp0s81lRjr\r\n" \
+    "F2AwIs/TLaMyMDAwDwYDVR0TAQH/BAUwAwEB/zAdBgNVHQ4EFgQUDf0KHNxzEy7q\r\n" \
+    "znA405Fx1lQsRLowCgYIKoZIzj0EAwIDSAAwRQIhAPDKNTxO8sLkns1y7ec2w2oR\r\n" \
+    "CYoQyDj2d498XeWYkSVuAiBz+GSRnTmdCFzQKfL8/ma7QaNdXihKYrWUdqvlynVV\r\n" \
+    "MQ==\r\n"                                                             \
+    "-----END CERTIFICATE-----\r\n"
+
+namespace ot {
+
+namespace Cli {
+
+otTcatVendorInfo sVendorInfo;
```

Comment:
Yes, it currently does.  This will go away if we decide to use vendor information from diagnostics.
To be decided in Threadgroup if TCAT requires diagnostics or if vendor information in TCAT is optional.

Reply:
I think we decided to require diagnostics, and stop using 0x24 Get Provisioning TLVs.

Reply:
next step: resolve this in a future PR (that removes the unneeded Provisioning TLVs and uses diagnostics instead)